packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -146,5 +147,24 @@ describe("Controller helper integration test", function () {
       expect(vaultBefore.shortAmount.add(mintWSqueethAmount).eq(vaultAfter.shortAmount)).to.be.true
       expect(depositorBalanceAfter.gt(depositorBalanceBefore.sub(value))).to.be.true
     })
+
+    it("flash close short position and buy long", async () => {
+      const vaultId = (await shortSqueeth.nextId()).sub(1);
+      await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address)
+
+      const vaultBefore = await controller.vaults(vaultId)
+      const longBalanceBefore = await wSqueeth.balanceOf(depositor.address)
+      const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+      const squeethToBuy = vaultBefore.collateralAmount.div(squeethPrice)
+
+      await controllerHelper.connect(depositor).flashswapWBurnBuyLong(vaultId, vaultBefore.shortAmount, squeethToBuy, vaultBefore.collateralAmount, vaultBefore.collateralAmount);
```

Comment:
I think we should do something similar with slippage that is done in flash-mint to calculate the maxToPay param.

I also think for both of them we need to check that the amount returned to the user is what we expect.  (But believe you want to do this in a separate PR)